tests/dialects/test_snowflake.py

```diff
@@ -1464,90 +1464,69 @@ def test_match_recognize(self):
 
     def test_show(self):
         # Parsed as Show
-        self.validate_identity("SHOW PRIMARY KEYS")
-        self.validate_identity("SHOW PRIMARY KEYS IN ACCOUNT")
-        self.validate_identity("SHOW PRIMARY KEYS IN DATABASE")
-        self.validate_identity("SHOW PRIMARY KEYS IN DATABASE foo")
-        self.validate_identity("SHOW PRIMARY KEYS IN TABLE")
-        self.validate_identity("SHOW PRIMARY KEYS IN TABLE foo")
-        self.validate_identity(
-            'SHOW PRIMARY KEYS IN "TEST"."PUBLIC"."customers"',
-            'SHOW PRIMARY KEYS IN TABLE "TEST"."PUBLIC"."customers"',
-        )
-        self.validate_identity(
-            'SHOW TERSE PRIMARY KEYS IN "TEST"."PUBLIC"."customers"',
-            'SHOW PRIMARY KEYS IN TABLE "TEST"."PUBLIC"."customers"',
-        )
-        self.validate_identity(
-            "show terse schemas in database db1 starts with 'a' limit 10 from 'b'",
-            "SHOW TERSE SCHEMAS IN DATABASE db1 STARTS WITH 'a' LIMIT 10 FROM 'b'",
-        )
-        self.validate_identity(
-            "show terse objects in schema db1.schema1 starts with 'a' limit 10 from 'b'",
-            "SHOW TERSE OBJECTS IN SCHEMA db1.schema1 STARTS WITH 'a' LIMIT 10 FROM 'b'",
-        )
-        self.validate_identity(
-            "show terse objects in db1.schema1 starts with 'a' limit 10 from 'b'",
-            "SHOW TERSE OBJECTS IN SCHEMA db1.schema1 STARTS WITH 'a' LIMIT 10 FROM 'b'",
-        )
-        self.validate_identity(
-            "SHOW TABLES LIKE 'line%' IN tpch.public",
-            "SHOW TABLES LIKE 'line%' IN SCHEMA tpch.public",
-        )
-        self.validate_identity(
-            "SHOW TABLES HISTORY IN tpch.public",
-            "SHOW TABLES HISTORY IN SCHEMA tpch.public",
-        )
-        self.validate_identity(
-            "show terse tables in schema db1.schema1 starts with 'a' limit 10 from 'b'",
-            "SHOW TERSE TABLES IN SCHEMA db1.schema1 STARTS WITH 'a' LIMIT 10 FROM 'b'",
-        )
-        self.validate_identity(
-            "show terse tables in db1.schema1 starts with 'a' limit 10 from 'b'",
-            "SHOW TERSE TABLES IN SCHEMA db1.schema1 STARTS WITH 'a' LIMIT 10 FROM 'b'",
-        )
-
-        ast = parse_one('SHOW PRIMARY KEYS IN "TEST"."PUBLIC"."customers"', read="snowflake")
-        table = ast.find(exp.Table)
-
-        self.assertEqual(table.sql(dialect="snowflake"), '"TEST"."PUBLIC"."customers"')
+        show_kinds = [
```

Comment:
So I dont know if this is what you were expecting or not. Most of each show variant has slightly different supported options, defaults and whatnot. (I can undo this commit and let you refactor it yourself, if this isn't what you were going for)

I couldn't see a way to make the tests more compact with a loop, other than by categorizing each differing variant and using that to choose what to do.

Arguably this is a decent amount less easy to read/follow for each individual option, but it's certainly more compact.

---

I guess also relevant, is that the parser doesn't currently distinguish between the different options in a bunch of ways. The availability of `IN TABLE` as an option, `HISTORY` being a supported option, etc.

Those seem like preexisting issues that might one day be worth revising (or not, if we only care about the max syntax), but i figured i'd point it out.

But as such, I just opted to not test those variants for the options that dont appear to support them IRL.

Reply:
I see what you mean, maybe you're right that splitting them across different methods could be easier to read / follow. It seems that's what we already do in [MySQL's case](https://github.com/tobymao/sqlglot/blob/main/tests/dialects/test_mysql.py).

Apologies for the back and forth, but if possible let's revert to the previous commit like you suggested (keep the constant, though), and I can clean this up by refactoring the existing `test_show` as well.

Appreciate the quick turnaround on this :)

Reply:
no problem! done!